### DISCUSSION TITLE
run the current .rs without moving cursor to codelens

### DISCUSSION THREAD

**Discussion by ssh352** [mrcjkb/rustaceanvim#328] [Q&A] [answered]
currently I need to move cursor to codelens line to Run.

Is it possible to Run anywhere? just like in an IDE? Thanks!

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

There's currently no such feature.

- You can use `:RustLsp runnables|debuggables|testables`, which will prompt you for a target,
  and `:RustLsp! runnables|debuggables|testables` will rerun the previous target, if one exists.
- In my personal neovim config, I have [added a module for navigating code lenses](https://github.com/mrcjkb/nvim/blob/288134b21e8fd24a99b368de33d8c3de43d9d8b9/lib/lua/mrcjk/lsp/codelens.lua) in the same way you can navigate diagnostics.
  [Here](https://github.com/mrcjkb/nvim/blob/288134b21e8fd24a99b368de33d8c3de43d9d8b9/nvim/after/ftplugin/rust.lua?plain=1#L67) is an example for a keymap I have set up that uses it.
  
It might be possible for me to add a command that runs the closest target to the cursor.
I've created [an issue](https://github.com/mrcjkb/rustaceanvim/issues/329) for it.
